dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -17,74 +17,132 @@
+            It defaults to ``DummyOutcomeRefuter.DEFAULT_TRUE_CAUSAL_EFFECT``, which means that there is no relationship between the treatment and outcome in the
+            dummy data.
+
+            The equation for the dummy outcome is given by
+            ``y_hat = h(t) + f(W)``
+            
+            where
+
+            * ``y_hat`` is the dummy outcome 
+            * ``h(t)`` is the function that gives the true causal effect 
+            * ``f(W)`` is the best estimate of ``y`` obtained keeping ``t`` constant. This ensures that the variation in output of function ``f(w)`` is not caused by `t`.
+
+            .. note:: The true causal effect should take an input of the same shape as the treatment and the output should match the shape of the outcome
+
+        required_variables (int, list, bool) : The list of variables to be used as the input for ``y~f(W)``
+            This is ``True`` by default, which in turn selects all variables leaving the treatment and the outcome
+
+            1. An integer argument refers to how many variables will be used for estimating the value of the outcome
+            2. A list explicitly refers to which variables will be used to estimate the outcome
+               Furthermore, it gives the ability to explictly select or deselect the covariates present in the estimation of the 
+               outcome. This is done by either adding or explicitly removing variables from the list as shown below: 
+            
+                For example:
+
+                We need to pass required_variables = ``[W0,W1]`` if we want ``W0`` and ``W1``.
+                
+                We need to pass required_variables = ``[-W0,-W1]`` if we want all variables excluding ``W0`` and ``W1``.
+            
+            3. If the value is True, we wish to include all variables to estimate the value of the outcome.
+               A False value is INVALID and will result in an error. 
+            
+            .. note:: These inputs are fed to the function for estimating the outcome variable. The same set of required_variables is used for each
+                instance of an internal estimation function.
+
+        bucket_size_scale_factor (float): For continuous data, the scale factor helps us scale the size of the bucket used on the data.
+            The default scale factor is ``DummyOutcomeRefuter.DEFAULT_BUCKET_SCALE_FACTOR``
+            ::
+
+                The number of buckets is given by: 
+                    (max value - min value)
+                    ------------------------
+                    (scale_factor * std_dev)
+
+        min_data_point_threshold (int): The minimum number of data points for an estimator to run.
+            This defaults to ``DummyOutcomeRefuter.MIN_DATA_POINT_THRESHOLD``. If the number of data points is too few
+            for a certain category, we make use of the ``DummyOutcomeRefuter.DEFAULT_TRANSFORMATION`` for generaring the dummy outcome            
```

Comment:
this is odd. What if there is one category for which there is insufficient data, but all others have sufficient data? Then one of them has a default simple transformation, whereas others have a complicated one. So then I think the warning should also include which category has less number of data points, and repeat the warning for each category.

Reply:
That makes sense.